patch can be dropped once all supported releases have
https://www.illumos.org/issues/14534 integrated

diff -wpruN '--exclude=*.orig' a~/test/test_flac.sh a/test/test_flac.sh
--- a~/test/test_flac.sh	1970-01-01 00:00:00
+++ a/test/test_flac.sh	1970-01-01 00:00:00
@@ -1206,7 +1206,7 @@ flac2flac ()
 	run_metaflac --list out.flac | filter > out1.meta || die "ERROR listing metadata of output FLAC file"
     # Ignore lengths which can be affected by the version string.
     sed "s/length:.*/length: XXX/" out1.meta > out.meta
-	diff -q -w $expect out.meta 2>/dev/null || die "ERROR: metadata does not match expected $expect"
+	diff -w $expect out.meta >/dev/null 2>&1 || die "ERROR: metadata does not match expected $expect"
 	echo OK
 }
 
